compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -424,13 +424,14 @@ fn run_compiler(
             }
 
             // Make sure name resolution and macro expansion is run.
-            queries.global_ctxt()?.enter(|tcx| tcx.resolver_for_lowering());
+            let mut gcx = queries.global_ctxt()?;
```

Comment:
This change was attempted before and reverted - https://github.com/rust-lang/rust/pull/107740.

Reply:
Huh, we should add a comment to point this out linking to #107740 and why.

Reply:
FWIW https://github.com/rust-lang/rust/pull/132410 changes after_analysis to pass in TyCtxt directly, avoiding the double locking issue.